test/integration/extensions_test.go

```diff
@@ -0,0 +1,229 @@
+package integration
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	testutil "github.com/openshift/origin/test/util"
+)
+
+func TestExtensions(t *testing.T) {
+	// Create a temporary directory.
+	tmpdir, err := ioutil.TempDir("", "extensions")
+	if err != nil {
+		t.Fatalf("Could not create tmp dir for extensions: %v", err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	// Create extension files.
+	masterOptions := setupExtensions(t, tmpdir)
```

Comment:
I like having it as a separate function because it's more obvious where setup ends and the actual tests begin. The function gets really long without it.